spec/support/database_cleaner.rb

```diff
@@ -0,0 +1,31 @@
+RSpec.configure do |config|
+  config.before(:suite) do
+    DatabaseCleaner.clean_with(:truncation)
+    Test.setup_models
+    Rails.application.load_seed
+  end
+
+  config.before(:each) do
+    DatabaseCleaner.strategy = :transaction
+  end
+
+  config.before(:each, js: true) do
+    DatabaseCleaner.strategy = :truncation
```

Comment:
I think it's one of those cases where it is better to be conservative and always do it, rather than be bitten later by trying too hard to finely parse whether you need it or not. The reason we truncate is that capybara is operating in a different process and transactions can't be assured across processes.